server/cache.go

```diff
@@ -225,6 +225,13 @@ func (r *regionsInfo) randFollowerRegion(storeID uint64) *regionInfo {
 	return randRegion(r.followers[storeID])
 }
 
+func (r *regionsInfo) randomRegion() *regionInfo {
+	for _, region := range r.regions {
```

Comment:
You are right. We should not rely on the randomness of map iteration. I'm working on another PR to correct it.